### PR TITLE
Split cache into files

### PR DESCRIPTION
**Pull request by avsetsin**

### CODE REVIEW

File: src/cache/cache.service.ts [hunk 1]
```diff
@@ -44,34 +50,89 @@ export class CacheService<T extends unknown> {
...
+  private async getCacheFilePaths(): Promise<string[]> {
     const dir = await this.getCacheDirPath();
-    return join(dir, this.cacheFile);
+    const existCacheFiles: string[] = [];
+
+    let batchIndex = 0;
+
+    do {
+      try {
+        const filePath = join(dir, this.getCacheFileName(batchIndex));
```
Comment (eddort) [inline src/cache/cache.service.ts:65]: Perhaps we should use glob to find the files we need and exclude others?

File: src/cache/cache.service.ts [hunk 2]
```diff
@@ -44,34 +50,89 @@ export class CacheService<T extends unknown> {
...
+      const to = (batchIndex + 1) * this.cacheBatchSize;
+      const batchedData = data.slice(from, to);
+
+      if (batchedData.length === 0) return;
+
+      const filePath = join(dirPath, this.getCacheFileName(batchIndex));
+      await writeFile(filePath, JSON.stringify({ headers, data: batchedData }));
+
+      batchIndex++;
+    } while (true);
```
Comment (eddort) [inline src/cache/cache.service.ts:129]: I would suggest using a stricter loop, without the "while true".
And we need to consider the data integrity scenario. I suggest specifying the number of chunks in the headers.

**Comment by infloop** [comment]
why we need force push here?

File: src/cache/cache.service.ts [hunk 3]
```diff
@@ -1,40 +1,46 @@
...
+      this.cache = await this.getCacheFromFiles();
     }
 
     return this.cache;
   }
 
   public async setCache(cache: T): Promise<void> {
     this.cache = cache;
-    return await this.saveCacheToFile();
+    return await this.saveCacheToFiles();
```
Comment (infloop) [inline src/cache/cache.service.ts:38]: Maybe add removal of cache files before actual saving of new files?

File: src/cache/cache.service.ts (hunk 2 above)
Comment (avsetsin) [reply src/cache/cache.service.ts]: replaced with a for loop

File: src/cache/cache.service.ts (hunk 3 above)
Comment (avsetsin) [reply src/cache/cache.service.ts:39]: done in `saveCacheToFiles` method

File: src/cache/cache.service.ts (hunk 1 above)
Comment (avsetsin) [reply src/cache/cache.service.ts]: replaced with `glob`

**Review by infloop — APPROVED** [review]

**Pull request merged**